docs/admin/architectures/index.md

```diff
@@ -0,0 +1,343 @@
+
+This pattern closely reflects how our customers naturally use the system. SSH
+connections are heavily utilized because they're the primary communication
+channel for IDEs with VS Code and JetBrains plugins.
+
+The basic setup of scale tests environment involves:
+
+1. Scale tests runner (32 vCPU, 128 GB RAM)
+2. Coder: 2 replicas (4 vCPU, 16 GB RAM)
+3. Database: 1 instance (2 vCPU, 32 GB RAM)
+4. Provisioner: 50 instances (0.5 vCPU, 512 MB RAM)
+
+The test is deemed successful if users did not experience interruptions in their
+workflows, `coderd` did not crash or require restarts, and no other internal
+errors were observed.
+
+### Traffic Projections
+
+In our scale tests, we simulate activity from 2000 users, 2000 workspaces, and
+2000 agents, with two items of workspace agent metadata being sent every 10
+seconds. Here are the resulting metrics:
+
+Coder:
+
+- Median CPU usage for _coderd_: 3 vCPU, peaking at 3.7 vCPU during dashboard
+  tests.
+- Median API request rate: 350 RPS during dashboard tests, 250 RPS during Web
+  Terminal and workspace apps tests.
+- 2000 agent API connections with latency: p90 at 60 ms, p95 at 220 ms.
+- on average 2400 Web Socket connections during dashboard tests.
+
+Provisionerd:
+
+- Median CPU usage is 0.35 vCPU during workspace provisioning.
+
+Database:
+
+- Median CPU utilization is 80%, with a significant portion dedicated to writing
+  metadata.
+- Memory utilization averages at 40%.
+- `write_ops_count` between 6.7 and 8.4 operations per second.
+
+## Available reference architectures
+
+[Up to 1,000 users](1k-users.md)
+
+[Up to 2,000 users](2k-users.md)
+
+[Up to 3,000 users](3k-users.md)
+
+## Hardware recommendation
+
+### Control plane: coderd
+
+To ensure stability and reliability of the Coder control plane, it's essential
+to focus on node sizing, resource limits, and the number of replicas. We
+recommend referencing public cloud providers such as AWS, GCP, and Azure for
+guidance on optimal configurations. A reasonable approach involves using scaling
+formulas based on factors like CPU, memory, and the number of users.
+
+While the minimum requirements specify 1 CPU core and 2 GB of memory per
+`coderd` replica, it is recommended to allocate additional resources depending
+on the workload size to ensure deployment stability.
+
+#### CPU and memory usage
+
+Enabling [agent stats collection](../../cli.md#--prometheus-collect-agent-stats)
+(optional) may increase memory consumption.
+
+Enabling direct connections between users and workspace agents (apps or SSH
+traffic) can help prevent an increase in CPU usage. It is recommended to keep
+[this option enabled](../../cli.md#--disable-direct-connections) unless there
+are compelling reasons to disable it.
+
+Inactive users do not consume Coder resources.
+
+#### Scaling formula
+
+When determining scaling requirements, consider the following factors:
+
+- `1 vCPU x 2 GB memory x 250 users`: A reasonable formula to determine resource
```

Comment:
I think this better matches our reference arch?

```suggestion
- `0.5 vCPU x 2 GB memory x 250 users`: A reasonable formula to determine resource
```

Reply:
Thinking about the future, I would leave `1 vCPU`, WDYT?

Reply:
We can probably argue for wiggle room here based on how certain Terraform providers may be more CPU-intensive than others.